### ISSUE TITLE
jar to app with args

### ISSUE THREAD

**Issue by mfouka** [Jorl17/jar2app#33]
I would like to pack a java application with script
my script starts with java -jar myapp.jar file.txt
to convert it to macos
I converted my jar with your tools but the problem when I run
open -n myapp.app file.txt the program does not see the arguments?

**Comment by veronicakulikova** [comment]
Aren't you supposed to use --args before your arguments to pass them to the osx app?

**Comment by nfouka** [comment]
We Can change the name of the JavaAppLauncher script?
We have to do something like that
java -jar myapp.jar $ 1?
or fill in the <**args**>?

**Comment by veronicakulikova** [comment]
JavaAppLauncher (the default name) is the app's stub executable that by default just runs the jar as if by `java -jar whatever.jar`. In place of `<args>` in your case would be `file.txt`
The full invocation would be `myapp.app/Contents/MacOS/JavaAppLauncher file.txt`

**Comment by veronicakulikova** [comment]
That's not what --executable-name is intended for. It's for being able to rename the standard JavaAppLauncher stub to whatever name you want the built app to use. If you were to name it run.sh, I'd imagine the name would be applied but wouldn't execute properly as having a shell script be the main executable of a .app bundle is... Odd.
Try building the app as you originally did but run it with `open -n -a myapp.app --args file.txt`
This should be the most straightforward way. Tell me if that doesn't work, my MacBook's in the shop so I can't really test this.

**Comment by nfouka** [comment]
it works ! Thank you very much for your details !

**Comment by nfouka** [comment]
I tested with command  open -n -a myapp.app --args file.txt , but it does not work!

**Comment by veronicakulikova** [comment]
What do you get in the terminal, and does the app launch

**Comment by nfouka** [comment]
The application does not launch, no message on the console !

**Comment by nfouka** [comment]
I tested on high sierra and it opens with the command open myapp.app --args file.txt (open without option)
On the other hand when I open my file from the desktop and I double click (or open with myapp.app) my application starts but she still does not see the file
do you have an idea ?
Thank you

**Comment by nfouka** [comment]
Yes exactly, but without **parameters**!
I tested it on Sierre Height and El Capian 
Thank You

**Comment by veronicakulikova** [comment]
Heh. Added PR #34, which you should be able to use with the likes of [universalJavaApplicationStub](https://github.com/tofi86/universalJavaApplicationStub) to get arguments working.

Once I used jar2app from that PR with `-x universalJavaApplicationStub` (universalJavaApplicationStub downloaded from the repo linked above was in the same directory as the jar I packaged), I was able to get arguments working as expected.
<img width="457" alt="screen shot 2018-11-29 at 6 41 44 pm" src="https://user-images.githubusercontent.com/10523871/49267391-180d1f00-f40f-11e8-8ee8-cfed5788353d.png">
<img width="442" alt="screen shot 2018-11-29 at 7 41 53 pm" src="https://user-images.githubusercontent.com/10523871/49267396-1e9b9680-f40f-11e8-8109-0be798a0c13b.png">

I doubt that the PR will get merged any time soon (given both splash PRs have been idle for ages) so you can just download [my fork's branch](https://github.com/veronicakulikova/jar2app/tree/custom-stub) if you want.

**Comment by Jorl17** [comment]
Hi!

As you can tell, I don't at the moment have very much time to give the support I want to in any of my projects, but I've merged the pending PRs. Do you confirm they are working?

If you need anything else, do say, I'll try to find some free time between work and side-project work to fit in gratis-side-project work :)

Thank you for your developments!

**Comment by Lucina** [comment]
Yes, I tested the user-provided stub PR with multiple cases on my programs (no customization, custom name only, custom stub only [name is kept to default JavaAppLauncher], custom name + custom stub [stub is copied with a new name). It works as you'd expect.
Seriously, thanks for making this amazing project.

**Comment by nfouka** [comment]
Great Work ! 
Thank You.

**Comment by Lucina** [comment]
Is your issue resolved / do you want to close this issue?
